src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/versioncontrol/CommitMessageToCommentTranscriber.java

```diff
@@ -83,26 +80,37 @@ public void postComment(ApplicationUser user, RevCommit commit) {
 		}
 	}
 
-	public void postComments() throws PermissionException {
-		ApplicationUser defaultUser;
-		try {
-			defaultUser = ComponentAccessor.getUserManager().createUser(DEFAULT_COMMIT_COMMENTATOR_USR_DETAILS);
-		} catch (CreateException e) {
-			defaultUser = ComponentAccessor.getUserManager().getUserByName(DEFAULT_COMMIT_COMMENTATOR_USR_NAME);
-		}
+	public void postComments(Ref branch) throws PermissionException {
+		ApplicationUser defaultUser = getUser();
 		String projectKey = this.issue.getProjectObject().getKey();
 		if (Boolean.parseBoolean(ConfigPersistenceManager.getValue(projectKey, "isPostFeatureBranchCommitsActivated"))) {
+
+			Optional.ofNullable(gitClient.getFeatureBranchCommits(branch)).ifPresent(featureBranchCommits::addAll);
 			for (RevCommit commit : this.featureBranchCommits) {
-				this.postComment(defaultUser, commit);
+				this.postComment(defaultUser, commit, branch);
 			}
 		}
+		if (branch.getName().contains("develop") || branch.getName().contains("master")) {
+			if (Boolean.parseBoolean(ConfigPersistenceManager.getValue(projectKey, "isPostSquashedCommitsActivated"))) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://app.codacy.com/manual/UHD/cures-condec-jira/pullRequest?prid=4848760)